slack_bolt/slack_function/slack_function.py

```diff
@@ -78,8 +77,94 @@ def __call__(*args, **kwargs):
 
         return __call__
 
-    # TODO add view listener
+        # -------------------------
+
+    # view
+
+    def view(
+        self,
+        constraints: Union[str, Pattern, Dict[str, Union[str, Pattern]]],
+        matchers: Optional[Sequence[Callable[..., bool]]] = None,
+        middleware: Optional[Sequence[Union[Callable, Middleware]]] = None,
+    ) -> Callable[..., Optional[Callable[..., Optional[BoltResponse]]]]:
+        """Registers a new `view_submission`/`view_closed` event listener for a function.
+
+            # Define a function handler using the function decorator
+            @app.function("function_1")
+            def sample_view(event, complete: Complete):
+                client.views_open(
+                    interactivity_pointer=interactivity_pointer,
+                    view={...}
+                )
+
+            # Use this method as a decorator
+            @my_function.view("view_1")
+            def handle_submission(ack, body, client, view, complete):
+                # Acknowledge the view_submission event and close the modal
+                ack()
```

Comment:
You need to move this line to the part below the input validation part. Meaning that the current code snippet calls `ack()` twice in the case of input errors.